src/EditorFeatures/Core/Implementation/ForegroundNotification/ForegroundNotificationService.cs

```diff
@@ -139,6 +147,15 @@ private void NotifyOnForegroundWorker()
                         {
                             // eat up cancellation
                         }
+                        catch (Exception ex) when (FatalError.ReportWithoutCrash(ex))
+                        {
+                            // The PendingWork callbacks should never throw.  In the case they do we
+                            // must ensure the IAsyncToken implementation is completed.  If it is not
```

Comment:
Two issues with this:

1. The code now can't decide if it's trying to ensure we cleanly recover from this state or not. The comment says "callbacks should never throw", but we're now trying to clean up which means we're sure expecting it. Should we just ReportWithCrash() this? It seems easier and safer than us trying to march along in a state we clearly never considered.
2. The comment seems off to me: it's not just unit tests that will get screwed up by this. You're calling out a very specific symptom here of a very large problem. This feels like saying "we need to make sure there's gas in the car, because it's important to keep the low fuel light off." Sure, yeah, that's one thing it fixes, but not the only one...